packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -260,8 +285,6 @@ class AnalyzeCommand extends FlutterCommand {
 
     int exitCode = await process.exitCode;
 
-    host.deleteSync(recursive: true);
```

Comment:
On or around line 295 (`source.readAsLinesSync()`) we read the source code for files with analysis errors. If that includes the generated stub entrypoint file, you get an exception (which is what I was seeing). We were deleting the source before the read -